api/namespace.go

```diff
@@ -100,10 +100,21 @@ func (n *Namespaces) Update(ns *Namespace, q *WriteOptions) (*Namespace, *WriteM
 
 func (n *Namespaces) Read(name string, q *QueryOptions) (*Namespace, *QueryMeta, error) {
 	var out Namespace
-	qm, err := n.c.query("/v1/namespace/"+name, &out, q)
+	r := n.c.newRequest("GET", "/v1/namespace/"+name)
+	r.setQueryOptions(q)
+	rtt, resp, err := requireOK(n.c.doRequest(r))
```

Comment:
Did you also want to handle non-200 responses, like interpret a 404 and send back a `nil` for the `*Namespace` return parameter?

Reply:
We already will return Nil for a non-200 status code. The `requireOk` will translate a non-200 into an error and the below err check ends up returning nil.

Should we maybe return no error at all for a 404?

Reply:
Sometimes when I'm using the API i find myself wishing for more "successful errors" like having a `404` return both a `nil` item and a `nil` error. With existing methods it's tricky to fix after the fact if it changes how the method operates which is behaviorally not really backwards compatible. At least with new methods you have the option of considering the alternative.

Reply:
I see you implemented a `requireNotFoundOrOk` for use with role management. I can switch it to use that.